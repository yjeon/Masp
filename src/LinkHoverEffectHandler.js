// Link.react.js
import React, { Component } from 'react';

const STATUS = {
  NORMAL: 'normal',
  HOVERED: 'hovered',
};

class LinkHoverEffectHandler extends Component {
  constructor() {
    super();

    this._onMouseEnter = this._onMouseEnter.bind(this);
    this._onMouseLeave = this._onMouseLeave.bind(this);

    this.state = {
      class: STATUS.NORMAL,
    };
  }

  _onMouseEnter() {
    this.setState({class: STATUS.HOVERED});
  }

  _onMouseLeave() {
    this.setState({class: STATUS.NORMAL});
  }

  render() {
    return (
      <a
        classname={this.state.class}
        href={this.props.page || '#'}
        onMouseEnter={this._onMouseEnter}
        onMouseLeave={this._onMouseLeave}>
        {this.props.children}
      </a>
    );
  }
}

export default LinkHoverEffectHandler;
