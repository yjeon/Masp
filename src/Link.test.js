import React from 'react';
import renderer from 'react-test-renderer';
import LinkHoverEffectHandler from './LinkHoverEffectHandler';

test('Link changes the class when hovered', () => {
    const component = renderer.create(
      <LinkHoverEffectHandler page="http://www.facebook.com">Facebook</LinkHoverEffectHandler>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseEnter();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseLeave();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
