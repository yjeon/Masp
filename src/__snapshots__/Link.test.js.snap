exports[`test Link changes the class when hovered 1`] = `
<a
  classname="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function bound _onMouseEnter]}
  onMouseLeave={[Function bound _onMouseLeave]}>
  Facebook
</a>
`;

exports[`test Link changes the class when hovered 2`] = `
<a
  classname="hovered"
  href="http://www.facebook.com"
  onMouseEnter={[Function bound _onMouseEnter]}
  onMouseLeave={[Function bound _onMouseLeave]}>
  Facebook
</a>
`;

exports[`test Link changes the class when hovered 3`] = `
<a
  classname="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function bound _onMouseEnter]}
  onMouseLeave={[Function bound _onMouseLeave]}>
  Facebook
</a>
`;
